### DISCUSSION TITLE
Create Sqlite db if it does not exist

### DISCUSSION THREAD

**Discussion by AndiLeni** [loupe-php/loupe#39] [Ideas]
Currently, I have to create the sqlite database myself, otherwise I receive an error when I call the `create` function of the LoupeFactory.
I tried doing this via `$db = new SQLite3("loupe.db"));`
This however creates an empty database. When I call `search` on this database before adding any documents, an error is thrown.
This could be handled better in my opinion, since it is not so uncommon to not have any documents in the database at the beginning.
Would it be possible to have something like a `setup` function, which creates a new sqlite batabase with the required tables or at least populates an existing sqlite database so it is usable?

**Comment by Toflar** [comment]
Yes, the goal is that you create the file yourself. You don't need `SQLite3`, you just have to make sure the empty file exists. A simple `file_put_contents('loupe.db', '')` is enough. While this usually just works, there are quite some quirks if the parent directory does not etc. which is why developers usually rely on libraries like `symfony/filesystem` or others. See https://github.com/symfony/filesystem/blob/6.3/Filesystem.php#L659-L694 for example.

I did not implement this by design because developers usually already have one of those file handling libraries installed and I didn't want to add another `require` in the `composer.json` just for that case.
